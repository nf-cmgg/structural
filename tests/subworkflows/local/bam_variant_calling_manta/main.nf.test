nextflow_workflow {

    name "Test Workflow BAM_VARIANT_CALLING_MANTA"
    script "subworkflows/local/bam_variant_calling_manta/main.nf"
    workflow "BAM_VARIANT_CALLING_MANTA"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "bam_variant_calling_manta"

    config "./nextflow.config"

    test("homo_sapiens") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.cram1, checkIfExists: true),
                    file(params.crai1, checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                input[3] = Channel.value([
                    file(params.manta_config, checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path(workflow.out.manta_vcfs[0][1]).linesGzip[-4..-1],
                    file(workflow.out.manta_vcfs[0][2]).name,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

}
