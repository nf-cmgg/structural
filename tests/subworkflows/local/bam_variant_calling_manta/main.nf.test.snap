{
    "homo_sapiens": {
        "content": [
            [
                "##contig=<ID=chr16,length=2000001>",
                "##contig=<ID=chrX,length=2000001>",
                "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tPosCon1",
                "chr16\t1061455\tmanta_DEL_1\tGTTCCACTCATGTTGTTGCAGATTACTGGATCTCACTCTTTTTTATGGGTGAATCGTACTTCATAGTGTATATGTACC\tG\t619\tPASS\tCALLERS=manta;CIPOS=0,5;END=1061532;SVLEN=-77;SVTYPE=DEL\tGT:PE:SR\t1/1:0,0:0,19"
            ],
            "test.svync.manta.vcf.gz.tbi",
            6,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-11T14:12:51.499950856"
    }
}